test/e2e/e2e.test.ts

```diff
@@ -358,3 +358,84 @@ describe('#end-to-end keyword replacement', function () {
     recordingDone();
   });
 });
+
+describe('keyword preservation', () => {
+  const config = {
+    AUTH0_DOMAIN,
+    AUTH0_CLIENT_ID,
+    AUTH0_CLIENT_SECRET,
+    AUTH0_ACCESS_TOKEN,
+    AUTH0_PRESERVE_KEYWORDS: true,
+    AUTH0_INCLUDED_ONLY: ['tenant', 'emailTemplates'] as AssetTypes[],
+    AUTH0_KEYWORD_REPLACE_MAPPINGS: {
+      TENANT_NAME: 'This tenant name should be preserved',
+      DOMAIN: 'travel0.com',
+      LANGUAGES: ['en', 'es'],
+    },
+  };
+
+  it('should preserve keywords for yaml format', async function () {
+    const workDirectory = testNameToWorkingDirectory(this.test?.title);
+
+    const { recordingDone } = await setupRecording(this.test?.title);
+
+    await deploy({
+      input_file: `${__dirname}/testdata/should-preserve-keywords/yaml/tenant.yaml`,
+      config,
+    });
+
+    copySync(`${__dirname}/testdata/should-preserve-keywords/yaml`, workDirectory); //It is necessary to copy directory contents to work directory to prevent overwriting of Git-committed files
+
+    await dump({
+      output_folder: workDirectory,
+      format: 'yaml',
+      config,
+    });
+
+    const yaml = yamlLoad(fs.readFileSync(path.join(workDirectory, 'tenant.yaml')));
+    expect(yaml.tenant.friendly_name).to.equal('##TENANT_NAME##');
+    expect(yaml.tenant.support_email).to.equal('support@##DOMAIN##');
+    expect(yaml.tenant.support_url).to.equal('https://##DOMAIN##/support');
+    // expect(yaml.tenant.enabled_locales).to.equal('@@LANGUAGES@@'); TODO: enable @@ARRAY@@ keyword preservation in yaml formats
+
+    // const emailTemplateHTML = fs
+    //   .readFileSync(path.join(workDirectory, 'emailTemplates', 'welcome_email.html'))
+    //   .toString();
+    // expect(emailTemplateHTML).to.contain('##TENANT##'); TODO: enable keyword preservation in auxillary template files
```

Comment:
This and the similar test below will be added in the coming days when the relevant fix has been added.